.ci/openshift-ci/run_smoke_test.sh

```diff
@@ -40,13 +40,25 @@ hello_msg='Hello World'
 oc exec ${pod} -- sh -c "echo $hello_msg > $hello_file"
 
 info "Creating the service and route"
-oc apply -f ${script_dir}/smoke/service.yaml
-sleep 60
+if oc apply -f ${script_dir}/smoke/service.yaml; then
+    # Likely on OCP, use service
+    is_ocp=1
+    sleep 60
+    host=$(oc get route/http-server-route -o jsonpath={.spec.host})
+    port=80
+else
+    # Likely on plain kubernetes, test using another container
+    is_ocp=0
+    info "Failed to create service, likely not on OCP, trying via NodePort"
+    oc apply -f "${script_dir}/smoke/service_kubernetes.yaml"
```

Comment:
Any reason for not waiting 60 seconds like in the OCP case ?

Reply:
OCP uses service+route which might take a while to establish reliably while here we use nodeport directly. Anyway on top I added the `retry` so it would be possible to skip the 60s timeout on OCP as well as `curl` should handle that. I can add that as another commit if you want to.